digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/IntegrationApi.scala

```diff
@@ -155,6 +161,78 @@ class IntegrationApi(val massTransitStopService: MassTransitStopService, implici
+      val roadNumber = lanes.head.attributes("ROAD_NUMBER")
+      val lanesGroupedByLaneCode = lanes.groupBy(_.laneAttributes.find(_.publicId == "lane_code")).values
+
+      val lanesWithContinuing = lanesGroupedByLaneCode.map(laneGroup => {
+        laneGroup.map(lane => {
+          val roadIdentifier = roadLinks(lane.linkId).roadIdentifier
+          val continuingLanes = LanePartitioner.getContinuingWithIdentifier(lane, roadIdentifier, laneGroup, roadLinks, true)
+          LaneWithContinuingLanes(lane, continuingLanes)
+        })
+      }).toSeq
+
+
+      val connectedGroups = lanesWithContinuing.flatMap(laneGroup => {
+        val startingLanes = getStartingLanes(laneGroup)
+        val connectedLanes = startingLanes.map(startingLane => {
+          getConnectedLanes(Seq(startingLane), laneGroup).sortBy(_.lane.id)
+        }).distinct
+        connectedLanes.map(_.map(_.lane))
+      })
+
+      val velhoLanes = connectedGroups.map(group => {
+        val laneCode = group.head.laneAttributes.find(_.publicId == "lane_code").get.values.head.value.asInstanceOf[Int]
+        val startAddressM = group.map(_.attributes("START_ADDR")).asInstanceOf[Seq[Long]].min
+        val endAddressM = group.map(_.attributes("END_ADDR")).asInstanceOf[Seq[Long]].max
+        VelhoLane(laneCode, startAddressM, endAddressM)
+      })
+
+      RoadPart(roadNumber, roadPartNumber, velhoLanes)
+    }).toSeq
+
+    roadNumbers.map(roadNumber => {
+      val roadParts = allRoadParts.filter(_.roadNumber == roadNumber)
+      val roadPartsWithOutRoadNumber = roadParts.map(_.copy(roadNumber = None))
+      VelhoRoad(roadNumber, roadPartsWithOutRoadNumber)
+    })
+  }
+
+  def lanesToApi(municipalityNumber: Int): Seq[Map[String, Any]] = {
+    val roadLinksInMunicipality = roadLinkService.getRoadLinksFromVVH(municipalityNumber)
+    val lanes = laneService.getLanesByRoadLinks(roadLinksInMunicipality)
+
+    val (lanesWithViiteAddress, noRoadAddress) = roadAddressService.laneWithRoadAddress(Seq(lanes)).flatten.partition(_.attributes.contains("VIITE_ROAD_NUMBER"))
+    val lanesWithTempAddress =  roadAddressService.experimentalLaneWithRoadAddress(Seq(noRoadAddress)).flatten.filter(_.attributes.contains("TEMP_ROAD_NUMBER"))
+    val lanesWithRoadAddress = lanesWithViiteAddress ++ lanesWithTempAddress
+
+    val lanesWithNormalRoadAddress = lanesWithRoadAddress.map(lane => {
+      val roadNumber = lane.attributes.getOrElse("VIITE_ROAD_NUMBER", "TEMP_ROAD_NUMBER")
+      val roadPartNumber = lane.attributes.getOrElse("VIITE_ROAD_PART_NUMBER", "TEMP_ROAD_PART_NUMBER")
+      val startAddr = lane.attributes.getOrElse("VIITE_START_ADDR", "TEMP_START_ADDR")
+      val endAddr = lane.attributes.getOrElse("VIITE_END_ADDR", "TEMP_END_ADDR")
+      lane.copy(attributes = lane.attributes + ("ROAD_NUMBER" -> roadNumber, "ROAD_PART_NUMBER" -> roadPartNumber,
+        "START_ADDR" -> startAddr, "END_ADDR" -> endAddr))
```

Comment:
Toimiiko tämä varmasti oikein? Eli yrittää ensin hakea esim "VIITE_ROAD_NUMBER" ja jos ei löydy niin sitten hakee temp arvon eikä aseta merkkijonoa "TEMP_ROAD_NUMBER" arvoksi? Entä miten me käsitellään tilanne jos kumpaakaan arvoa ei löydy?
Muualla (esim. [digiroad2api](https://github.com/finnishtransportagency/digiroad2/blob/76572f7f210fb0193d3178ad1e705e0787db6bb3/digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/Digiroad2Api.scala#L722)) laitettu näin: 

> roadLink.attributes.getOrElse("VIITE_ROAD_PART_NUMBER", roadLink.attributes.get("TEMP_ROAD_PART_NUMBER"))

Reply:
Nyt on tullut tässä hieno aivopieru, korjaan